src/containerapp/azext_containerapp/containerapp_env_certificate_decorator.py

```diff
@@ -0,0 +1,166 @@
+    NAME_ALREADY_EXISTS, NAME_INVALID
+from ._client_factory import handle_non_404_exception, handle_raw_exception
+from ._models import ContainerAppCertificateEnvelope as ContainerAppCertificateEnvelopeModel
+
+logger = get_logger(__name__)
+
+
+class ContainerappEnvCertificateDecorator(BaseResource):
+    def get_argument_location(self):
+        return self.get_param("location")
+
+    def get_argument_thumbprint(self):
+        return self.get_param("thumbprint")
+
+    def get_argument_certificate(self):
+        return self.get_param("certificate")
+
+    def get_argument_managed_certificates_only(self):
+        return self.get_param("managed_certificates_only")
+
+    def get_argument_private_key_certificates_only(self):
+        return self.get_param("private_key_certificates_only")
+
+    def check_cert_name_availability(self, resource_group_name, name, cert_name):
+        name_availability_request = {"name": cert_name, "type": CHECK_CERTIFICATE_NAME_AVAILABILITY_TYPE}
+        try:
+            return self.client.check_name_availability(self.cmd, resource_group_name, name, name_availability_request)
+        except CLIError as e:
+            handle_raw_exception(e)
+
+    def get_private_certificates(self, certificate_name=None):
+        if certificate_name:
+            try:
+                r = self.client.show_certificate(self.cmd, self.get_argument_resource_group_name(), self.get_argument_name(), certificate_name)
+                return [r] if certificate_matches(r, self.get_argument_location(), self.get_argument_thumbprint()) else []
+            except Exception as e:
+                handle_non_404_exception(e)
+                return []
+        else:
+            try:
+                r = self.client.list_certificates(self.cmd, self.get_argument_resource_group_name(), self.get_argument_name())
+                return list(filter(lambda c: certificate_matches(c, self.get_argument_location(), self.get_argument_thumbprint()), r))
+            except Exception as e:
+                handle_raw_exception(e)
+
+    def get_managed_certificates(self, certificate_name=None):
+        if certificate_name:
+            try:
+                r = self.client.show_managed_certificate(self.cmd, self.get_argument_resource_group_name(), self.get_argument_name(), certificate_name)
+                return [r] if certificate_location_matches(r, self.get_argument_location()) else []
+            except Exception as e:
+                handle_non_404_exception(e)
+                return []
+        else:
+            try:
+                r = self.client.list_managed_certificates(self.cmd, self.get_argument_resource_group_name(), self.get_argument_name())
+                return list(filter(lambda c: certificate_location_matches(c, self.get_argument_location()), r))
+            except Exception as e:
+                handle_raw_exception(e)
+
+
+class ContainerappEnvCertificateListDecorator(ContainerappEnvCertificateDecorator):
```

Comment:
should it be PreviewDecorator?

Reply:
Got it, I split List to GA and Preview Decorator.